lib/mkCargoDerivation.nix

```diff
@@ -54,9 +75,11 @@ let
     else null;
   cargoLock = args.cargoLock or cargoLockFromContents;
 in
-chosenStdenv.mkDerivation (cleanedArgs // lib.optionalAttrs (cargoLock != null) {
-  inherit cargoLock;
-} // {
+chosenStdenv.mkDerivation (
+  lib.optionalAttrs (!(args.noCrossToolchainEnv or false) && pkgs.buildPlatform != pkgs.hostPlatform) (mkCrossToolchainEnv stdenvSelector)
```

Comment:
Maybe we can move the `buildPlatform != hostPlatform` check to inside of `mkCrossToolchainEnv` (have it return `{}` if they are the same) just to simplify the logic every caller has to copy

Reply:
Resolved in fd19b86.